### PR TITLE
Feature - Data model for Blog app

### PR DESCRIPTION
**Pull request by wickathou**
# Features
- [x] Added models for posts, likes and comments
- [x] Added scaffold for users
- [x] Added relationships between tables
- [x] App using PostgreSQL as DB
- [x] Added tailwindcss

### CODE REVIEW

**Review by Davidosky007 — COMMENTED** [review]
Hi @wickathou 👋, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that doesn't follow the proper Gitflow pattern, kinldy create a pull request to compare your feature branch(data model) with the development branch(default branch)  and submit the link to that pull request via your Dashboard.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: db/migrate/20231025082805_create_posts.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class CreatePosts < ActiveRecord::Migration[7.0]
+  def change
+    create_table :posts do |t|
+      t.references :user, null: false, foreign_key: true
+      t.string :title
+      t.string :text
+      t.integer :commentsCounter
```
Comment (thecodechaser) [inline db/migrate/20231025082805_create_posts.rb:7]: - [x]  Column holding the foreign key to the `users` table should be named `author_id` in the `posts` table as requested in the project requirements, Please make sure to fix this. It should be like this:
```ruby 
t.references :author, null: false, foreign_key: {to_table: 'users'}, index: true
```

image

File: db/migrate/20231025083203_create_likes.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class CreateLikes < ActiveRecord::Migration[7.0]
+  def change
+    create_table :likes do |t|
+      t.references :user, null: false, foreign_key: true
+      t.references :post, null: false, foreign_key: true
```
Comment (thecodechaser) [inline db/migrate/20231025083203_create_likes.rb:5]: - [x] Please make sure that all columns that are foreign keys should have a corresponding index as requested in the project requirements
 - It should be like:
```ruby
t.references :user, null: false, foreign_key: true, index: true
t.references :post, null: false, foreign_key: true, index: true
```

File: db/migrate/20231025083234_create_comments.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+class CreateComments < ActiveRecord::Migration[7.0]
+  def change
+    create_table :comments do |t|
+      t.references :user, null: false, foreign_key: true
+      t.references :post, null: false, foreign_key: true
```
Comment (thecodechaser) [inline db/migrate/20231025083234_create_comments.rb:5]: - [x] Please make sure that all columns that are foreign keys should have a corresponding index as requested in the project requirements
 - It should be like:
```ruby
t.references :user, null: false, foreign_key: true, index: true
t.references :post, null: false, foreign_key: true, index: true
```

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @wickathou👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
- Good job with data model. ✔️
- No linter errors. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- PR has a good title and summary. ✔️



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by wickathou** [comment]
Issues fixed with migration
https://github.com/wickathou/mcv-blog/blob/data-model/db/migrate/20231025113002_rename_references.rb

**Review by mukhammadrizooff — APPROVED** [review]
Hi @wickathou, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

 

## Highlights
- No linter errors ✔️ 
- Correct GitFlow ✔️ 
- Good implementations ✔️
- Descriptive `README` file ✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**